tools/cldr-apps/src/main/webapp/js/new/cldrLoad.js

```diff
@@ -336,2471 +423,2246 @@ const cldrLoad = (function () {
+          error: errFunction,
+        };
+        cldrAjax.queueXhr(xhrArgs);
+      }
+    } else if (cldrStatus.getCurrentSpecial() == "none") {
+      // for now - redirect
+      cldrSurvey.hideLoader();
+      isLoading = false;
+      window.location = cldrStatus.getSurvUrl(); // redirect home
+    } else if (cldrStatus.getCurrentSpecial() == "locales") {
+      cldrSurvey.hideLoader();
+      isLoading = false;
+      var theDiv = document.createElement("div");
+      theDiv.className = "localeList";
+
+      addTopLocale("root", theDiv);
+      // top locales
+      for (var n in locmap.locmap.topLocales) {
+        var topLoc = locmap.locmap.topLocales[n];
+        addTopLocale(topLoc, theDiv);
+      }
+      flipper.flipTo(pages.other, null);
+      cldrEvent.filterAllLocale(); // filter for init data
+      cldrEvent.forceSidebar();
+      cldrStatus.setCurrentLocale(null);
+      cldrStatus.setCurrentSpecial("locales");
+      cldrSurvey.showInPop2(
+        cldrText.get("localesInitialGuidance"),
+        null,
+        null,
+        null,
+        true
+      ); /* show the box the first time */
+      $("#itemInfo").html("");
+    } else {
+      otherSpecial.show(cldrStatus.getCurrentSpecial(), {
+        flipper: flipper,
+        pages: pages,
+      });
+    }
+  } // end shower
+
+  /**
+   * Update the #hash and menus to the current settings.
+   *
+   * @param doPush {Boolean} if false, do not add to history
+   */
+  function updateHashAndMenus(doPush) {
+    const sessionId = cldrStatus.getSessionId();
+    const surveyUser = cldrStatus.getSurveyUser();
+    const userID = surveyUser && surveyUser.id ? surveyUser.id : 0;
+    const surveyUserPerms = cldrStatus.getPermissions();
+    const surveyUserURL = {
+      myAccountSetting: "survey?do=listu",
+      disableMyAccount: "lock.jsp",
+      recentActivity: "myvotes.jsp?user=" + userID + "&s=" + sessionId,
+      xmlUpload: "upload.jsp?a=/cldr-apps/survey&s=" + sessionId,
+      manageUser: "survey?do=list",
+      flag: "tc-flagged.jsp?s=" + sessionId,
+      about: "about.jsp",
+      browse: "browse.jsp",
+      adminPanel: "SurveyAjax?what=admin_panel&s=" + sessionId,
+    };
+
+    /**
+     * 'name' - the js/special/___.js name
+     * 'hidden' - true to hide the item
+     * 'title' - override of menu name
+     */
+    var specialItems = new Array();
+    if (surveyUser != null) {
+      specialItems = [
+        {
+          divider: true,
+        },
 
-                var theDiv = flipper.flipToEmpty(pages.other); // clean slate, and proceed..
+        {
+          title: "Admin Panel",
+          url: surveyUserURL.adminPanel,
+          display: surveyUser && surveyUser.userlevelName === "ADMIN",
+        },
+        {
+          divider: true,
+          display: surveyUser && surveyUser.userlevelName === "ADMIN",
+        },
 
-                removeAllChildNodes(theDiv);
+        {
+          title: "My Account",
+        }, // My Account section
```

Comment:
nice. I'll look forward to seeing this.


Any thought of splitting out `specialItems[]` to a separate file? I know it has code in it, but that could be abstracted as an input.

Reply:
I'm thinking of moving most menu-related code from cldrLoad.js into a new cldrMenu.js, then sorting out the dependencies. It also sounds like a good idea to further separate between code and data. Also, the gear menu maybe shouldn't act like a pull-down menu; clicking the gear icon could simply open a page with "#gear" in the url.